module-2/backend/week_three.md

```diff
@@ -10,14 +10,57 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What is the entry at the command line to create a new rails app?
+
+`rails new [project_name] -T --database=postgresql --skip-turbolinks --skip-spring`
+
 2. What do Models generally inherit from in rails?
+
+Models inherit ApplicationRecord
+
 3. What do Controllers generally inherit from in a rails project?
+
+Controllers inherit ApplicationController
+
 4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+
+`resources :horses only: :show`
+
 5. What rake task is useful when looking at routes, and what information does it give you?
+
+`rake routes` shows a list of all routes that your application can currently be pointed towards 
+
 6. What is an example of a route helper? When would you use them?
+
+`<%= link_to "More Info", horse_path(@horse) %>` Used when you need a way of accessing a route, and is used instead of hardcoding the paths into the view. 
+
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+
+`_path` is root relative, and `_url` is an absoulute path
+
 8. What are strong params and why are the necessary?
+
+Strong params allow us to only take in information that we want to receive, they help protect against SQL injection.
+
 9. What role does `form_for` play in helping us create our forms?
+
+It allows us to easily build forms with the correct HTTP verbs associated automatically.
+
 10. How does `form_for` know where to submit the user's input?
+
+The instance variable that is passed from the controller.
+
 11. Create a form using a `form_for` helper to create a new `Horse`. 
+
+```<%= form_for @horse do |f| %>
+  <%= f.label :name %>
+  <%= f.text_field :name %>
+  
+  <%= f.label :age %>
+  <%= f.number_field :age %>
+  
+  <%= f.submit %>
+<% end %>```
+
 12. Why do we want to validate our models?
+
+To ensure that all necessary information is received when creating/updating our database.
```

Comment:
to validate presence, uniqueness or format are all good reasons for validations!